### PR TITLE
Point to the CITATION file in terra

### PR DESCRIPTION
**Pull request by 1ucian0**
Counter part of https://github.com/Qiskit/qiskit-terra/pull/10111 (`on hold` until that one is merged). For context, see https://github.com/Qiskit/RFCs/blob/master/0011-repo-rename.md

### CODE REVIEW

File: Qiskit.bib
Comment (garrison) [inline Qiskit.bib:1]: There is also a link to this file in `docs/faq.rst`.

File: README.md [hunk 1]
```diff
@@ -89,9 +89,9 @@ Now you're set up and ready to check out our
 
 ## Authors and Citation
 
-Qiskit is the work of many people who contribute to the project at
+Qiskit is the work of [many people who contribute to the project](https://github.com/Qiskit/qiskit-terra/graphs/contributors) at
 different levels. If you use Qiskit, please cite as per the included
-[BibTeX file](Qiskit.bib).
+[BibTeX file](https://github.com/Qiskit/qiskit-terra/blob/main/CITATION.md).
```
Comment (garrison) [inline README.md]:
```suggestion
[BibTeX file](https://github.com/Qiskit/qiskit-terra/blob/main/CITATION.bib).
```

File: CITATION.md [hunk 2]
```diff
@@ -0,0 +1 @@
+See https://github.com/Qiskit/qiskit-terra/blob/main/CITATION.md
```
Comment (garrison) [inline CITATION.md]:
```suggestion
See https://github.com/Qiskit/qiskit-terra/blob/main/CITATION.bib
```

File: Qiskit.bib
Comment (1ucian0) [reply Qiskit.bib:1]: The current link in `docs/faq.rst` looks broken anyways. So good chance to update. Thanks!

**Review by garrison — APPROVED** [review]

**Comment by Eric-Arellano** [comment]
No longer on hold. https://github.com/Qiskit/qiskit-terra/pull/10111 merged.

**Review by jakelishman — APPROVED** [review]
For posterity: we also needed to update the bot's configuration to stop trying to regenerate this file, which was done in Qiskit/qiskit-bot#35.

**Pull request merged**